File: Programming Basics with Python/По-сложни проверки/operations_between_numbers.py

num1 = int(input())
num2 = int(input())
operator = input()

if num2 == 0:
    print(f"Cannot divide {num1} by zero")
    quit()
elif operator == "+":
    result = num1 + num2
elif operator == "-":
    result = num1 - num2
elif operator == "*":
    result = num1 * num2
elif operator == "/":
    result = num1 / num2
elif operator == "%":
    result = num1 % num2

even_or_odd = ""
if (operator == "+" or operator == "-" or operator == "*") and num2 != 0:
    if result % 2 == 0:
        even_or_odd = "even"
    else:
        even_or_odd = "odd"
    
    print(f"{num1} {operator} {num2} = {result} - {even_or_odd}")
elif operator == "/":
    print(f"{num1} {operator} {num2} = {result:.2f}")
elif operator == "%" and num2 != 0:
     print(f"{num1} {operator} {num2} = {result}")

File: Programming Basics with Java/More complex checks/OperationsBetweenNumbers.java
import java.util.Scanner;

public class OperationsBetweenNumbers 
{
    public static void main(String[] args) 
    {
        Scanner input = new Scanner(System.in);

        int num1 = Integer.parseInt(input.nextLine());
        int num2 = Integer.parseInt(input.nextLine());
        String operator = input.nextLine();

        double result = 0.0;
        String num_is = "";

        switch (operator)
        {
            case "+":
            {
                result = num1 + num2;
                num_is = (result % 2 == 0) ? "even" : "odd";
                 break;
            }
            case "-":
            {
                result = num1 - num2;
                num_is = (result % 2 == 0) ? "even" : "odd";
                 break;
            }
            case "*":
            {
                result = num1 * num2;
                num_is = (result % 2 == 0) ? "even" : "odd";
                 break;
            }
            case "/":
            {
                result = (double) num1 / num2;
                 break;
            }
            case "%":
            {
                result = (double) num1 % num2;
                 break;
            }
        }

        if (num2 == 0 && (operator.equals("/") || operator.equals("%")))
        {
            System.out.printf("Cannot divide %d by zero", num1);
        }
        else if (operator.equals("/"))
        {
            System.out.printf("%d / %d = %.2f", num1, num2, result);
        }
        else if (operator.equals("%"))
        {
            System.out.printf("%d %% %d = %.0f", num1, num2, result);
        }
        else
        {
            System.out.printf("%d %s %d = %.0f - %s", num1, operator, num2, result, num_is);
        }
    }
}
